examples/s4u/synchro-semaphore/s4u-synchro-semaphore.cpp

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2010-2017. The SimGrid Team.
+ * All rights reserved.                                                     */
+
+/* This program is free software; you can redistribute it and/or modify it
+ * under the terms of the license (GNU LGPL) which comes with this package. */
+
+ #include "simgrid/s4u.hpp"
+ #include "xbt/str.h"
+ #include "simgrid/plugins/load.h"
+ #include "simgrid/simix.h"
+ #include <cstdlib>
+ #include <iostream>
+
+XBT_LOG_NEW_DEFAULT_CATEGORY(msg_semaphore_example, "Messages specific for this msg example");
+
+smx_sem_t sem;
```

Comment:
ARGH! This is an internal data structure, and the user should not see it. You need a s4u::Semaphore cleanly encapsulating all this before you can convert this example. Please see the s4u::Mutex for an example.